frontend/assets/style/components/_pills.scss

```diff
@@ -1,9 +1,29 @@
 .pill {
-  background: $text_0;
-  color: $general_2;
   font-weight: 600;
   border-radius: 3px;
-  padding: 0.375rem 0.5rem;
+  padding: 0.375rem $spacer-sm;
   line-height: 1;
   margin: 0 $spacer-xs;
+  display: inline-block;
+  white-space: nowrap;
+
+  &.is-small {
+    padding: ($spacer / 8) $spacer-xs;
+    text-transform: uppercase;
+    font-size: $size_5;
+    font-weight: 100;
+  }
+
+  &.has-background-dark {
+    background: $text_0;
+    color: $general_2;
+  }
+
+  @each $name in $colors {
+    &--#{$name} {
```

Comment:
all the other components with modifiers, follow the naming convention `.is-#{$same}`, (ex: button.is-primary). I believe we should do the same here: `.pill.is-primary` instead of `.pill--primary` to keep consistency.

And there's no need to do `@extend .pill;` following that approach.

Reply:
I was sure you would say that.. I'm thinking to change the other way around not following the boostrap style, but BEM actually.. because I never been a big fan of having multiple class on top of each others where you depend on order of the css.

Reply:
Hum... Depend on order of the css? Here we can do .pill.is-primary or .is-primary.pill, it would work.
Or are you talking about when writing  the css? So far we didn’t have any issue with CSS order, have we?

Reply:
It's not because you didn't had issue that we should use a technique which is not bug proof.. has you probably know .pill.is-primary or .is-primary.pill is the same, that's not what i'm talking about.
If you have a generique class like .is-primary in your colors.css and you add this colors.css you're dependent of where the tabs.css is located in main.css
I agree with you that this won't be the case here because we redefine .is-primary in the tab class.. But that's just assumption and therefore less effective that the BEM solution.
So now the question is: do we want to improve or stick to bad habit?

Reply:
Another advantage of using BEM is separation of concern between markup and css. What do you reckon?

Reply:
Anyway probably a different PR to do that

Reply:
We always use `.is-primary` within the context of the component. We never use `.is-primary` has a standalone class, so we are never dependent on where the CSS is placed or what's its order. Which means, we are bug-proof as in BEM.

```
// .is-primary is always a modifier "suffixed" to its block.
// CSS will always output as excepted regardless of its order.

.button.is-primary { ... }

.tabs.is-primary { ... }

.pills.is-primary { ... }

// We never declare .is-primary alone
.is-primary { ... }
```

If there's something I'm still missing, we can continue this discussion in a new issue and refactor our approach in a new PR.